website/docs/transforms/rename.md

```diff
@@ -17,37 +17,137 @@ Add the following configuration to your Mesh config file:
+              to:
+                type: User
+                field: surname
+transforms:
+  rename:
+      mode: wrap # bare won't work here at all-sources (root) level, because you're not using merger-bare
+      renames:
+        - from:
+            type: Country
+            field: ISO-3166_Code
+          to:
+            type: Country
+            field: code
+```
 
+> ProTip: When you want to use `wrap`, you can omit the "mode" property since this is already applied by default.  
+
+### Bare
+Bare is a recent addition and works by applying transformations to the original schema. The handler generates a GraphQL schema and passes it to the rename transform. `bare` will rename types and fields as defined in the schema generated by your handler and will return an updated version of the same schema with the transformations applied.  
+The advantage of `bare` is that it commits the schema as part of the initialisation process only; then it won't have any runtime cost.  
+
+#### Restrictions
+Bare does provide performance improvements over `wrap`; however it has a main restriction: it needs to access the bare schema. Here are some reasons why this might not work:
+- Your data source is already in GraphQL format.  
+In this case `bare` won't work as it cannot update a native GraphQL schema. This is not the same as updating a "translated" GraphQL schema (e.g. from JSON-schema, OpenApi, SOAP, etc.).
+The suggestion in this case is to apply `wrap` transforms to your GraphQL data source and `bare` transforms to sources "translated" into GraphQL.
+
+- You are applying rename transform at all-sources (root) level.  
+This means that `bare` would be dealing with a stitched schema and not a bare schema anymore. If you do want to use `bare` at the root level your only choice is to opt into [merger-bare](https://graphql-mesh.com/docs/api/modules/merger-bare/), which lets transforms access the bare schemas; because it merges sources without wrapping them. This works when you don't have (or you take care of) conflicts between your sources.
+
+- You are mixing transforms that supports `bare` with transforms that don't.  
+Again, `bare` always need to access the bare schema. If you define other transforms that don't support `bare` mode, you will most likely have troubles, since those transforms will apply a wrapping layer which won't let `bare` transforms access the bare schema.
+In order to take advantage of `bare` performance improvement, the suggestion here is to apply `wrap` transforms at the all-sources (root) level and `bare` transforms within the data sources level.
+
+Example:
+```yaml
+sources:
+  - name: Countries
+    handler:
+      soap:
+        wsdl: http://webservices.../wso?WSDL
+  - name: Users
+    handler:
+      openapi:
+        source: https://api.../swagger.yaml
+    transforms:
+      rename:
+          mode: bare # bare is a great choice here, at the data source level, as it doesn't have any runtime cost
```

Comment:
If you don't want to talk about "cost", consider changing this to "implications" as well; for consistency.